osu.Game.Rulesets.Catch/Difficulty/CatchDifficultyCalculator.cs

```diff
@@ -34,7 +34,7 @@ public CatchDifficultyCalculator(Ruleset ruleset, WorkingBeatmap beatmap)
         {
         }
 
-        protected override DifficultyAttributes Calculate(IBeatmap beatmap, Mod[] mods, double timeRate)
+        protected override DifficultyAttributes Calculate(double upTo, IBeatmap beatmap, Mod[] mods, double timeRate)
```

Comment:
Putting this as the first parameter everywhere requires all usages to be updated. In this case, I'd move the upTo parameter to the end and assign it a default value (for example, `DifficultyAttributes Calculate(IBeatmap beatmap, Mod[] mods, double timeRate, double upTo = Double.PositiveInfinity)`).

This way it won't break existing usages and your diff will be smaller.